### PR TITLE
Record

### PR DESCRIPTION
**Pull request by Nick007J**

### CODE REVIEW

File: src/control/Record.cpp [hunk 1]
```diff
@@ -2,18 +2,523 @@
...
+	pState->rightX = UINT8_MAX / 2 * pCar->GetRight().x;
+	pState->rightY = UINT8_MAX / 2 * pCar->GetRight().y;
+	pState->rightZ = UINT8_MAX / 2 * pCar->GetRight().z;
+	pState->forwardX = UINT8_MAX / 2 * pCar->GetForward().x;
+	pState->forwardY = UINT8_MAX / 2 * pCar->GetForward().y;
+	pState->forwardZ = UINT8_MAX / 2 * pCar->GetForward().z;
+	pState->pos = pCar->GetPosition();
+	pState->velX = 1.0f * (UINT16_MAX / 2) * pCar->GetMoveSpeed().x;
+	pState->velY = 1.0f * (UINT16_MAX / 2) * pCar->GetMoveSpeed().y;
+	pState->velZ = 1.0f * (UINT16_MAX / 2) * pCar->GetMoveSpeed().z;
```
Comment (ShFil119) [inline src/control/Record.cpp:342]: maybe `pState` should use more `CVector`?

Comment (Nick007J) [reply src/control/Record.cpp:342]: these are 1 byte variables

File: src/control/Record.cpp [hunk 2]
```diff
@@ -330,16 +330,16 @@ void CRecordDataForChase::SaveOrRetrieveCarPositions(void)
...
+	pState->forwardX = INT8_MAX * pCar->GetForward().x;
+	pState->forwardY = INT8_MAX * pCar->GetForward().y;
+	pState->forwardZ = INT8_MAX * pCar->GetForward().z;
 	pState->pos = pCar->GetPosition();
-	pState->velX = 1.0f * (UINT16_MAX / 2) * pCar->GetMoveSpeed().x;
-	pState->velY = 1.0f * (UINT16_MAX / 2) * pCar->GetMoveSpeed().y;
-	pState->velZ = 1.0f * (UINT16_MAX / 2) * pCar->GetMoveSpeed().z;
+	pState->velX = 0.5f * INT16_MAX * pCar->GetMoveSpeed().x;
+	pState->velY = 0.5f * INT16_MAX * pCar->GetMoveSpeed().y;
+	pState->velZ = 0.5f * INT16_MAX * pCar->GetMoveSpeed().z;
```
Comment (ShFil119) [inline src/control/Record.cpp:342]: shouldn't 0.5f be removed?

**Comment by erorcun** [comment]
lgtm

File: src/control/Record.cpp [hunk 3]
```diff
@@ -2,18 +2,523 @@
...
+			CFileMgr::CloseFile(FId);
+			FId = CFileMgr::OpenFileForWriting("record.dat");
+			RecordingState = STATE_RECORD;
+		}
+	}
+	else {
+		RecordingState = STATE_PLAYBACK;
+	}
+	if (RecordingState == STATE_PLAYBACK) {
+		pDataBufferPointer = (uint8*)malloc(MEMORY_FOR_GAME_RECORD);
```
Comment (aap) [inline src/control/Record.cpp:43]: new

File: src/control/Record.cpp (hunk 2 above)
Comment (Nick007J) [reply src/control/Record.cpp:342]: it was a mistake in original PR

**Pull request merged**